src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -1462,17 +1378,51 @@ public void onUtterance(Utterance utterance) throws Exception {
       }
     }
   }
+  
+  /**
+   * This receiver can take a config published by another service and sync
+   * predicates from it
+   * @param cfg
+   */
+  public void onConfig(ServiceConfig cfg) {

```

Comment:
What's the use case for this?    In general, I like the idea that a service could listen for a config, but I don't understand your specific motivation here for this one...

Reply:
Trying to tame the chaos that was made by gluing Jython with ProgramAB and a bajillion global variables..
at least in this context its a step towards structured data, by providing easier access to config in aiml